Geometry_Engine/Query/ExternalEdges.cs

```diff
@@ -102,38 +108,74 @@ public static List<ICurve> ExternalEdges(this PolySurface surface)
         /**** Public Methods - Mesh                     ****/
         /***************************************************/
 
-        public static List<Line> ExternalEdges(this Mesh mesh)
+        [PreviousVersion("6.0", "BH.Engine.Geometry.Query.ExternalEdges(BH.oM.Geometry.Mesh)")]
+        [Description("Gets the external edge curves of a mesh as a set of lines. Extraction is done by finding all unique edges in the mesh.")]
+        [Input("mesh", "The mesh to extract external edge curves from.")]
+        [Input("filterByTopology", "If true, edges with unique topology are returned. If false, edges with unique geometry is returned. Toggle only renders different result for meshes with duplicate nodes.")]
+        [Input("tolerance", "Tolerance to be used for identifying duplicate edges. Only used if filterByTopology is false.")]
+        [Output("edges", "The external edge curves of the mesh.")]
+        public static List<Line> ExternalEdges(this Mesh mesh, bool filterByTopology = false, double tolerance = Tolerance.Distance)
         {
             if (mesh == null)
                 return null;
 
             if (mesh.Faces.Count < 1)
                 return null;
```

Comment:
I would consider returning empty list here - empty list of faces should return an empty list of edges?

Reply:
Sure, can get behind that